tests/UnitTests/PhasorDynamics/GenClassicalTests.hpp

```diff
@@ -0,0 +1,216 @@
+#include <iomanip>
+#include <iostream>
+#include <limits>
+#include <Model/PhasorDynamics/Bus/Bus.hpp>
+#include <Model/PhasorDynamics/Bus/BusInfinite.hpp>
+#include <Model/PhasorDynamics/SynchronousMachine/GenClassical/GenClassical.hpp>
+#include <Model/PhasorDynamics/SynchronousMachine/GenClassical/GenClassical.cpp>
+#include <Utilities/TestHelpers.hpp>
+#include <Utilities/Testing.hpp>
+
+namespace GridKit
+{
+  namespace Testing
+  {
+
+    template <class ScalarT, typename IdxT>
+    class GenClassicalTests
+    {
+    private:
+      using real_type = typename PhasorDynamics::Component<ScalarT, IdxT>::real_type;
+
+    public:
+      GenClassicalTests()  = default;
+      ~GenClassicalTests() = default;
+
+      TestOutcome constructor()
+      {
+        TestStatus success = true;
+
+        auto* bus = new PhasorDynamics::Bus<ScalarT, IdxT>(1.0, 0.0);
+
+        PhasorDynamics::Component<ScalarT, IdxT>* machine =
+            new PhasorDynamics::GenClassical<ScalarT, IdxT>(bus, 1);
+
+        success *= (machine != nullptr);
+
+        if (machine)
+        {
+          delete machine;
+        }
+        delete bus;
+
+        return success.report(__func__);
+      }
+
+      /**
+       * A test case to verify residual values
+       */
+      TestOutcome residual()
+      {
+        TestStatus success = true;
+
+        // classical generator parameters
+        real_type H{0.1};
+        real_type D{2.35};
+        real_type Ra{1.5};
+        real_type Xdp{4.5};
+        real_type pmech{5.0};
+        real_type ep{2.5};
+
+        ScalarT Vr1{2.0}; ///< Bus-1 real voltage
+        ScalarT Vi1{1.5}; ///< Bus-1 imaginary voltage
+
+        const ScalarT res0{-1128.973355292326}; /// first residual
+        const ScalarT res1{27.5625000000000};   /// second residual
+        const ScalarT res2{4.102511525891203};  /// third residual
+        const ScalarT res3{8.164018441425924};  /// fourth residual
+        const ScalarT res4{2.089603682931281};  /// fifth residual
+        const ScalarT res5{5.0};                /// fifth residual
+        const ScalarT res6{2.5};                /// fifth residual
+        const ScalarT tol = 0.000000000001;     // tolerance for comparing results
+

```

Comment:
Using magic numbers like this makes the test fragile and the tolerance seems to be chosen arbitrarily (`1e-12`). Also, this will work only if the scalar type is `double` or lower precision.

Reply:
The tolerance should be of order of machine precision for the scalar data type.